### PR TITLE
Move CHIPClusters.* to gen/ folders

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem

`src/controller/CHIPClusters.*` files should be part of the `gen/` folders.

 #### Summary of Changes
 * Move `CHIPClusters.*` files to `gen/` folders

As often, this PR is splitted into 2 commits. The first one contains the real changes, the second one is the result of running `./scripts/tools/zap_regen_all.py`

### CODE REVIEW

**Review by mspang — APPROVED** [review]

File: src/app/zap-templates/templates/app/CHIPClusters-src.zapt [hunk 1]
```diff
@@ -13,7 +13,7 @@ namespace Controller {
 // TODO(#4503): length should be passed to commands when byte string is in argument list.
 // TODO(#4503): Commands should take group id as an argument.
 
-{{#chip_server_clusters}}
+{{#chip_client_clusters}}
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/templates/app/CHIPClusters-src.zapt:16]: Why the changes from server to client here and further down?

Comment (vivien-apple) [reply src/app/zap-templates/templates/app/CHIPClusters-src.zapt:16]: Because now the files will be generated on a per example basis, and the commands are generated based on the list of `client` clusters selected by the example. For example, `chip-tool` has many `client` clusters, but no `server`. Same things for `iOS ChipTool` and the Python device controller.

The fact that we still use `chip_server_clusters` here is historical. At the origin this file was generated from the `all-clusters-app` configuration.

**Review by woody-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
@msandstedt @saurabhst ?

**Review by saurabhst — APPROVED** [review]

File: src/app/zap-templates/templates/app/CHIPClusters-src.zapt (hunk 1 above)
Comment (bzbarsky-apple) [reply src/app/zap-templates/templates/app/CHIPClusters-src.zapt:16]: Gotcha.

**Comment by erjiaqing** [comment]
@vivien-apple I found the CHIPClusters.cpp in Python gen/ is actually empty. So the python device controller does not work after this (so the cirque test is failing after this PR)

File: src/app/zap-templates/templates/app/CHIPClusters-src.zapt (hunk 1 above)
Comment (erjiaqing) [reply src/app/zap-templates/templates/app/CHIPClusters-src.zapt:16]: For controller, I guess we still need `chip_client_clusters`, since we are sending commands to the server and we need to encode the commands supported by **server**
